### ISSUE TITLE
new lint: warn if fallback match case only covers a single case

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#4488]
````rust
pub enum Enum {
    A,
    B,
}

pub fn a(e: Enum) -> i32 {
 match e {
     Enum::A => 1,
     _ => 2,
 }
}
````
In this code `Enum` can only have two values, `A` or `B`.
The `_ =>` match arm can only cover as single case.
The problem is that as soon as more variants get added to `Enum`, they will be eaten by the `_ => 2` match arm and the user won't possibly notice, it might alter the behaviour in strange ways.

A lint that warns if `_ =>` only covers a single option and suggests replacing it with explicitly that option,like so:
````rust
 match e {
     Enum::A => 1,
     Enum::B => 2,
 }
````
could help here.

**Comment by jplatte** [comment]
:-1: from me. I've written code like this and the fallback applying to new enum variants is exactly the intention. E.g. consider `is_*` style methods:

```rust
pub enum Enum {
    A,
    B,
}

impl Enum {
    pub fn is_a(&self) -> bool {
        match self {
            Self::A => true,
            _ => false,
        }
    }

    pub fn is_B(&self) -> bool {
        match self {
            Self::B => true,
            _ => false,
        }
    }
}
```

I consider this to be idiomatic code and would be annoyed if it would fire a lint just because the enum happens to only have two variants.

**Comment by matthiaskrgr** [comment]
Actually it looks like I just didn't find the wildcard_enum_match_arm lint... 😅
